File: examples/higher-order-functions/map.ts
function map<T, U>(
  arr: Array<T>,
  f: (item: T) => U
): Array<U> {
  const result = [];
  for (const item of arr) {
    result.push(f(item));
  }
  return result;
}

const newArr = map([1, 2, 3], n => n * 2);
console.log(newArr);

File: examples/higher-order-functions/map.swift
func map<T, U>(
  _ arr: [T],
  _ f: (T) -> U) -> [U] {
    var result: [U] = []
    for item in arr {
        result.append(f(item))
    }
    return result
}

let newArr = map(
  [1, 2, 3],
  { (n: Int) in n * 2 }
)
print(newArr)
